src/figure/bounding_box.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+
+from typing import Tuple
+
+
+class BoundingBox:
+    """Base class for all detectors"""
+
+    def __init__(self, coords: Tuple[int, int, int, int] = (0, 0, 0, 0), label: str = '', confidence: float = 0):
+        """
+        coords list[ left_top_x, left_top_y, right_bot_x, right_bot_y]
+        label - string
+        confidence - 0.xx 
+        """
+        self.coords = coords
+        self.label = label
+        self.confidence = confidence
+
+        self.__check_coords()
+
+    @classmethod
+    def from_ultralytics(cls, data, label_names: dict):
+        """Constructs a BoundingBox object from Ultralytics data"""
+        boxes = []
+        for i in range(data.data.shape[0]):
+            x1, y1, x2, y2 = map(int, data.xyxy[i].tolist())
+            label_id = int(data.cls[i].item())
+            label = label_names.get(label_id, 'Unknown')
+            confidence = data.conf[i].item()
+            boxes.append(cls((x1, y1, x2, y2), label, confidence))
+        return boxes
```

Comment:
Why that is returning list of some unknown type boxes? It's a bit confusing, because you would suspect to create BoudningBox object from this data.

Reply:
It returns a list of BoudningBox objects not an unknown type. You are right that it could return one object which would be more clear